enterprise/javaee.project/src/org/netbeans/modules/javaee/project/api/ant/ui/logicalview/AbstractLogicalViewProvider.java

```diff
@@ -514,7 +514,8 @@ public String getShortDescription() {
         public Image getIcon( int type ) {
             Image original = super.getIcon( type );
             if (isBroken()) {
-                return ImageUtilities.mergeImages(original, ProjectProperties.ICON_BROKEN_BADGE.getImage(), 8, 0);
+                return ImageUtilities.mergeImages(original,
+                    ImageUtilities.icon2Image(ProjectProperties.ICON_BROKEN_BADGE), 8, 0);
```

Comment:
why is this change needed?

the utility method will simply call getImage() anyway, no?

Reply:
There's some custom handling for certain Icon subclasses, but reading over the ImageUtilities code, it indeed seems that the change is unnecessary! mergeImages is smart enough to preserve vector graphics in these cases.

I will remove the changes around mergeImages.